input/script_desc.go

```diff
@@ -73,6 +73,9 @@ type TapscriptDescriptor interface {
 
 	// TapScriptTree returns the underlying tapscript tree.
 	TapScriptTree() *txscript.IndexedTapScriptTree
+
+	// Tree returns the underlying ScriptTree.
+	Tree() ScriptTree
```

Comment:
Q: is `TapScriptTree` a subset of `Tree`?

Reply:
No, the `*txscript.IndexedTapScriptTree` is the parsed leaves in the structure that can then build the root hash. The `ScriptTree` is something every `TapscriptDescriptor` implementation of the `ScriptDescriptor` has embedded, but before this change there was no easy way to access it.

Pulled this commit out of the PR, since it's kind of standalone.

Reply:
@guggero the `Tree()` method is used in `tapchannel.LeavesFromTapscriptScriptTree`.